src/app/services/work-item.service.ts

```diff
@@ -731,46 +731,40 @@ export class WorkItemService {
       });
   }
 
-  getForwardLinkTypes(workItem: WorkItem): Observable<any> {
-    return this.http.get(workItem.links.targetLinkTypes)
+  /**
+   * Usage: This function fetches all the work item link types
+   * Store it in an instance variable
+   *
+   * @return Promise of LinkType[]
+   */
+  getAllLinkTypes(workItem: WorkItem): Observable<any> {
+    let workItemLinkTypesUrl = this._currentSpace.links.self + '/workitemlinktypes';
+    return this.http.get(workItemLinkTypesUrl)
       .catch((error: Error | any) => {
         this.notifyError('Getting link meta info failed (forward).', error);
         return Observable.throw(new Error(error.message));
       });
   }
 
-  getBackwardLinkTypes(workItem: WorkItem): Observable<any> {
-    return this.http.get(workItem.links.sourceLinkTypes)
-      .catch((error: Error | any) => {
-        this.notifyError('Getting link meta info failed (backward).', error);
-        return Observable.throw(new Error(error.message));
-      });
-  }
-
   /**
    * Usage: This function fetches all the work item link types
    * Store it in an instance variable
    *
    * @return Promise of LinkType[]
    */
   getLinkTypes(workItem: WorkItem): Observable<Object> {
-    return Observable.forkJoin(
-      this.getForwardLinkTypes(workItem),
-      this.getBackwardLinkTypes(workItem)
-    )
-    .map(items => {
-      let linkTypes: Object = {};
-      linkTypes['forwardLinks'] = items[0].json().data;
-      linkTypes['backwardLinks'] = items[1].json().data;
-      return linkTypes;
-    })
-    .map((linkTypes: any) => {
-      return this.formatLinkTypes(linkTypes);
-    })
-    .catch((err) => {
-      console.log(err);
-      return Observable.of({});
-    });
+    return Observable.forkJoin(this.getAllLinkTypes(workItem))
+        .map(items => {
+          let linkTypes: Object = {};
+          linkTypes['forwardLinks'] = items[0].json().data;
+          linkTypes['backwardLinks'] = items[1].json().data;
```

Comment:
Now items is not an array of array, so it'll fail here. 
Items would be one array having all the link types.
How would you separate forward and backward link here?